src/pykiso/lib/auxiliaries/udsaux/uds_auxiliary.py

```diff
@@ -271,6 +306,31 @@ def tester_present_sender(self, period: int = 4) -> Iterator[None]:
             stop_event.set()
             sender.join()
 
+    def start_tester_present_sender(self, period: int) -> None:
+        """Start to continuously sends tester present messages via UDS
+
+        :param period: period in seconds to use for the cyclic sending of tester present
+        :param aux_alias: auxiliary's alias
+        """
+
+        self.sender = threading.Thread(
+            name="TesterPresentSender",
+            target=self._sender_run,
+            args=(period, self.sender_stop_event),
+        )
+        self.sender.start()
+
+    def stop_tester_present_sender(self) -> None:
+        """Stop to continuously sends tester present messages via UDS"""
+        try:
+            self.sender_stop_event.set()
+            self.sender.join()
+            self.sender_stop_event.clear()
+        except AttributeError:
```

Comment:
mhm.. why not if self.sender_stop_event.is_alive() ?

Reply:
ah ok,  But what happens when the user calls 
start_tester_present_sender    multiple times ?
Does the unstopped thread then run in the background ?

Reply:
Should not happened now